src/main/java/net/sf/jabref/gui/EntryTypeDialog.java

```diff
@@ -148,6 +172,95 @@ private JPanel createEntryGroupPanel(String groupTitle, Collection<EntryType> en
         return panel;
     }
 
+    private JPanel createIdFetcherPanel() {
+        JButton generateButton = new JButton(Localization.lang("Generate"));
+        JButton cancelButton = new JButton(Localization.lang("Cancel"));
+        JTextField idTextField = new JTextField("");
+        JComboBox<String> comboBox = new JComboBox<>();
+        EntryFetchers.getIdFetchers().forEach(fetcher -> comboBox.addItem(fetcher.getName()));
+        JLabel fetcherLabel = new JLabel("ID type"), idLabel = new JLabel("ID");
+
+        SwingWorker<Optional<BibEntry>, Void> fetcherWorker = new SwingWorker<Optional<BibEntry>, Void>() {
+            Optional<BibEntry> bibEntry = Optional.empty();
+            IdBasedFetcher fetcher = null;
+            String searchID = "";
+
+            @Override
+            protected Optional<BibEntry> doInBackground() throws Exception {
+                searchID = idTextField.getText().trim();
+                fetcher = EntryFetchers.getIdFetchers().get(comboBox.getSelectedIndex());
+                if (!searchID.isEmpty()) {
+                    try {
+                        bibEntry = fetcher.performSearchById(searchID);
+                    } catch (FetcherException e) {
+                        LOGGER.error("Error fetching from " + fetcher.getName(), e);
+                        JOptionPane.showMessageDialog(null, Localization.lang("Error while fetching from %0", fetcher.getName()), Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+                    }
+                }
+                dispose();
+                return bibEntry;
+            }
+
+            @Override
+            protected void done() {
+                if (isCancelled()) {
+                    JOptionPane.showMessageDialog(null, Localization.lang("%0 import canceled", fetcher.getName()),Localization.lang("Import canceled by user"), JOptionPane.INFORMATION_MESSAGE);
```

Comment:
According to http://stackoverflow.com/a/6204600/873282 this is not thread safe as it is not called in the EDT. Maybe, it is also not called at all, according to the last comment by [searchengine27](http://stackoverflow.com/users/3560873/searchengine27) at http://stackoverflow.com/a/6204600/873282. Did you test whether this dialog appears?